testproject/Assets/Tests/Runtime/MultiprocessRuntime/ExecuteStepInContext.cs

```diff
@@ -0,0 +1,289 @@
+                object defaultObj = null;
+                if (paramType.IsValueType)
+                {
+                    defaultObj = Activator.CreateInstance(paramType);
+                }
+
+                parametersToReturn[i] = defaultObj;
+            }
+
+            return parametersToReturn;
+        }
+
+        foreach (var contextType in typesWithContextMethods)
+        {
+            var allConstructors = contextType.GetConstructors();
+            if (allConstructors.Length > 1)
+            {
+                throw new NotImplementedException("Case not implemented where test has more than one contructor");
+            }
+
+            var instance = Activator.CreateInstance(contextType, allConstructors.Length > 0 ? GetParameterValuesToPass(allConstructors[0].GetParameters()) : null);
+            m_AllClientTestInstances.Add(instance); // keeping that instance so tests can use captured local attributes
+
+            List<MethodInfo> typeMethodsWithContextSteps = new List<MethodInfo>();
+            foreach (var method in contextType.GetMethods())
+            {
+                if (method.GetCustomAttributes(typeof(ExecuteStepInContext.MultiprocessContextBasedTestAttribute), true).Length > 0)
+                {
+                    typeMethodsWithContextSteps.Add(method);
+                }
+            }
+
+            foreach (var method in typeMethodsWithContextSteps)
+            {
+                var parametersToPass = GetParameterValuesToPass(method.GetParameters());
+                var enumerator = (IEnumerator)method.Invoke(instance, parametersToPass.ToArray());
+                while (enumerator.MoveNext()) { }
+            }
+        }
+
+        isRegistering = false;
+        hasRegistered = true;
+    }
+
+    /// <summary>
+    /// Executes an action with the specified context. This allows writing tests all in the same sequential flow,
+    /// making it more readable. This allows not having to jump between static client methods and test method
+    /// </summary>
+    /// <param name="actionContext">context to use. for example, should execute client side? server side?</param>
+    /// <param name="stepToExecute">action to execute</param>
+    /// <param name="ignoreTimeoutException">waits for timeout and just finishes step execution silently</param>
+    /// <param name="paramToPass">parameters to pass to action</param>
+    /// <param name="networkManager"></param>
+    /// <param name="waitMultipleUpdates"> waits multiple frames before allowing the execution to continue. This means ClientFinishedServerRpc must be called manually</param>
+    /// <param name="additionalIsFinishedWaiter"></param>
+    public ExecuteStepInContext(StepExecutionContext actionContext, Action<byte[]> stepToExecute, bool ignoreTimeoutException = false, byte[] paramToPass = default, NetworkManager networkManager = null, bool waitMultipleUpdates = false, Func<bool> additionalIsFinishedWaiter = null)
```

Comment:
nit
```suggestion
    public ExecuteStepInContext(StepExecutionContext actionContext, Action<byte[]> stepToExecuteAction, bool ignoreTimeoutException = false, byte[] paramToPass = default, NetworkManager networkManager = null, bool waitMultipleUpdates = false, Func<bool> additionalIsFinishedWaiterFunc = null)
```

Reply:
not a fan of putting types in parameter names. mouse hover and auto complete in your IDE are there to help. Would you also rename ignoreTimeoutExceptionBool?